### PR TITLE
[#43] [Frontend] As a user, I can see the website header

### PR DESCRIPTION
**Pull request by hoangmirs**
Close https://github.com/nimblehq/til-web/issues/43

## What happened 👀

- Add the app header
- Bring back the Random button

## Insight 📝

- Copy from the Nimble website

## Proof Of Work 📹

The preview: https://deploy-preview-50--nimble-til.netlify.app/til/

|Mobile|Desktop|
|------|---------|
|Notification_Center|Notification_Center|

### CODE REVIEW

File: src/assets/icons/shuffle-random-svgrepo-com.svg [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="iso-8859-1"?>
```
Comment (Lahphim) [inline src/assets/icons/shuffle-random-svgrepo-com.svg:1]: @hoangmirs I think we don't need this file, do we?

File: src/components/Header/index.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  return (
+    <header className="app-header">
+      <nav
+        className={`app-navigation-bar navbar ${
+          isMenuOpen ? 'app-navigation-bar--menu-open' : ''
+        }`}
+      >
+        <Link href="/">
+          <a
+            href="/"
```
Comment (Lahphim) [inline src/components/Header/index.tsx:28]:
```suggestion
```
@hoangmirs I think we can omit `href` from this child as we've already used in the `Link`

File: src/components/Header/index.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        </button>
+
+        <nav className="app-navigation-bar__navigation">
+          <ul className="app-header__menu list--unstyled">
+            {categories.map((category) => (
+              <li key={category.name} className="app-header__menu-item">
+                <Link href={category.href}>
+                  <a
+                    className="app-header__menu-link link"
+                    href={category.href}
```
Comment (Lahphim) [inline src/components/Header/index.tsx:52]:
```suggestion
```
This one also, I think we can omit that 🤔

File: src/components/Header/index.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+                  >
+                    {category.name}
+                  </a>
+                </Link>
+              </li>
+            ))}
+          </ul>
+
+          <a
```
Comment (Lahphim) [inline src/components/Header/index.tsx:61]: Don't we wrap this tag with `<Link>`?

File: src/components/Header/index.tsx (hunk 2 above)
Comment (hoangmirs) [reply src/components/Header/index.tsx:28]: Yes, we can omit it, but ESLint will raise errors there. 
You can check this issue: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/anchor-is-valid.md#case-i-use-nextjs-and-im-getting-this-error-inside-of-links

I went with the solution adding `passHref` attribute to Link components and fixed in https://github.com/nimblehq/til-web/pull/50/commits/2af18151cc87fc2dd5e5af94bd10c14fb14592ed

File: src/assets/icons/shuffle-random-svgrepo-com.svg (hunk 1 above)
Comment (hoangmirs) [reply src/assets/icons/shuffle-random-svgrepo-com.svg]: Good catch, removed in https://github.com/nimblehq/til-web/pull/50/commits/1b2c461245697b20450d76da467ae8cec392d7a8

File: src/components/Header/index.tsx (hunk 4 above)
Comment (hoangmirs) [reply src/components/Header/index.tsx]: Yes, we should wrap with it. Added

File: src/components/Header/index.tsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+const Header = () => {
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+
+  return (
+    <header className="app-header">
+      <nav
+        className={`app-navigation-bar navbar ${
+          isMenuOpen ? 'app-navigation-bar--menu-open' : ''
+        }`}
```
Comment (tyrro) [inline src/components/Header/index.tsx:24]: maybe we could simplify it using `classNames`?

```suggestion
        className={classNames('app-navigation-bar navbar',  { 'app-navigation-bar--menu-open' : isMenuOpen })}
```

**Review by tyrro — APPROVED** [review]

File: src/components/Header/index.tsx (hunk 2 above)
Comment (Lahphim) [reply src/components/Header/index.tsx:28]: 🥲

**Review by Lahphim — APPROVED** [review]

**Pull request merged**